app/controllers/admin/sponsors_controller.rb

```diff
@@ -8,7 +8,19 @@ def index
       authorize! :index, Sponsor.new(conference_id: @conference.id)
     end
 
-    def edit; end
+    def show
+      @sponsor.swag.reject! { |_key, value| value[:type].blank? || value[:quantity].blank? }
+      @sponsor.swag_transportation.reject! { |_key, value| value[:carrier_name].blank? || value[:tracking_number].blank? || value[:boxes].blank? }
+      @sponsor.swag_index = @sponsor.swag.length
```

Comment:
This variable is being used in 1 place only. Do we really have to set it here? Can't we just call the @sponsor.swag.length where we need it?